docs/articles/writing-tests.rst

```diff
@@ -0,0 +1,306 @@
+Writing Tests
+#############
+
+Each test that should have access to the remote hosts must be marked with one or
+more topology markers. This tells pytest-mh what domains, hosts and roles are
+required to run the test. The marker also defines how the
+:class:`~pytest_mh.MultihostRole` objects should be accessible from within the
+test.
+
+The recommended way is to use :ref:`"dynamic" fixtures
+<writing_tests_dynamic_fixtures>`, which are fixtures that do not exist anywhere
+in the code but are injected into the test parameters by pytest-mh. It is also
+possible to get the access through the :func:`~pytest_mh.mh` fixture, but this
+is quite low level and should be avoided, unless you have a valid use case for
+it.
+
+.. seealso::
+
+    The topology, topology marker and related information is deeply covered in
+    :doc:`extending/multihost-topologies`.
+
+Using the mh fixture - low-level API
+====================================
+
+.. warning::
+
+    Using the :func:`~pytest_mh.mh` fixture directly is supported but not
+    recommended. You should avoid that unless you have a valid use case for it.
+    It is, however, documented first, to get better understanding of how things
+    work.
```

Comment:
Do you mean `However, it is necessary to do some research first in order to better understand how things work.`? If not, I don't know what you mean by this phrase.

Reply:
Makes sense, adding it to the corrections.